splink/linker.py

```diff
@@ -2178,6 +2183,94 @@ def _compute_metrics_nodes(
+        df_node_metrics: SplinkDataFrame,
+        df_predict: SplinkDataFrame,
+        df_clustered: SplinkDataFrame,
+        threshold_match_probability: float,
+    ) -> SplinkDataFrame:
+        """
+        Internal function for computing edge-level metrics.
+
+        Accepts outputs of `linker._compute_node_metrics()`, `linker.predict()` and
+        `linker.cluster_pairwise_at_threshold()`, along with the clustering threshold
+        and produces a table of edge metrics.
+
+        Uses `igraph` under-the-hood for calculations
+
+        Edge metrics produced:
+        * is_bridge (is the edge a bridge?)
+
+        Output table has a single row per edge, and the metric is_bridge:
+        |-------------------------------------------------------------|
+        | composite_unique_id_l | composite_unique_id_r   | is_bridge |
+        |-----------------------|-------------------------|-----------|
+        | s1-__-10001           | s1-__-10003             | true      |
+        | s1-__-10001           | s1-__-10005             | false     |
+        | s1-__-10005           | s1-__-10009             | false     |
+        | s1-__-10021           | s1-__-10024             | true      |
+        ...
+        """
+        try:
+            import igraph as ig
+        except ImportError:
+            raise SplinkException(
+                "You need to install the 'igraph' package to compute "
+                "the edge metric 'is_bridge'."
+            ) from None
+        uid_cols = self._settings_obj._unique_id_input_columns
+        # need composite unique ids
+        composite_uid_edges_l = _composite_unique_id_from_edges_sql(uid_cols, "l")
+        composite_uid_edges_r = _composite_unique_id_from_edges_sql(uid_cols, "r")
+
+        # firstly we (arbitrarily) map node ids to 1-indexed integers with no gaps
+        # this is how igraph deals with nodes
+        sql_infos = _node_mapping_table_sql(df_node_metrics)
+        for sql_info in sql_infos:
+            self._enqueue_sql(**sql_info)
+        df_node_mappings = self._execute_sql_pipeline()
+
+        # we keep only edges at or above relevant threshold
+        sql_info = _truncated_edges_sql(df_predict, threshold_match_probability)
+        self._enqueue_sql(**sql_info)
+        df_truncated_edges = self._execute_sql_pipeline()
+
+        # we map the truncated edges to the integer encoding for nodes above,
+        # keeping only the list of endpoints
+        sql_info = _edges_for_igraph_sql(
+            df_node_mappings,
+            df_truncated_edges.physical_name,
+            composite_uid_edges_l,
+            composite_uid_edges_r,
+        )
+        self._enqueue_sql(**sql_info)
+        edges_for_igraph = self._execute_sql_pipeline()
+        # we will need to manually register a table, so we use the hash from this table
+        igraph_edges_hash = edges_for_igraph.physical_name[-9:]
+        df_edges_for_igraph = edges_for_igraph.as_pandas_dataframe()
```

Comment:
Could potentially cause performance issues if the edges table is very large.  Think it's probably fine though as it's only two columns and alternative approaches (e.g. 'pagination' type thing would be v complex)